### PR TITLE
Namgold/catch-faulty-data

### PR DESCRIPTION
**Pull request by namgold**
Signed-off-by: Nam Nguyen <nam.nguyen@kyber.network>

### CODE REVIEW

File: src/hooks/Tokens.ts [hunk 1]
```diff
@@ -248,12 +248,14 @@ export function searchInactiveTokenLists({
     if (!list) continue
     for (const tokenInfo of list.tokens) {
       if (tokenInfo.chainId === chainId && tokenFilter(tokenInfo)) {
-        const wrapped: WrappedTokenInfo = new WrappedTokenInfo(tokenInfo, list)
-        if (!activeTokens[wrapped.address] && !addressSet[wrapped.address]) {
-          addressSet[wrapped.address] = true
-          result.push(wrapped)
-          if (result.length >= minResults) return result
-        }
+        try {
```
Comment (viet-nv) [inline src/hooks/Tokens.ts:251]: why do you need try catch here?
just check: `if (isAddress(tokenInfo.address) && tokenInfo.chainId === chainId && tokenFilter(tokenInfo))`

Comment (namgold) [reply src/hooks/Tokens.ts:251]: But :pepe-sad: đâu ai nói trước chữ ngờ

File: src/state/lists/hooks.ts [hunk 2]
```diff
@@ -60,9 +60,10 @@ function listToTokenMap(list: TokenList): TokenAddressMap {
     if (tokenMap[tokenInfo.chainId][formattedAddress] !== undefined) {
       return tokenMap
     }
-
-    const token = new WrappedTokenInfo(tokenInfo, list)
-    tokenMap[tokenInfo.chainId][formattedAddress] = token
+    try {
+      const token = new WrappedTokenInfo(tokenInfo, list)
```
Comment (viet-nv) [inline src/state/lists/hooks.ts:64]: same here. tokenMap is guarantee that address is checksum. dont need try catch here

File: src/hooks/Tokens.ts (hunk 1 above)
Comment (viet-nv) [reply src/hooks/Tokens.ts:251]: requested change

**Review by viet-nv — CHANGES_REQUESTED** [review]

File: src/state/lists/hooks.ts (hunk 2 above)
Comment (namgold) [reply src/state/lists/hooks.ts]: This is main place where crash happended. The above `src/hooks/Tokens.ts` is just an additional.

**Review by viet-nv — APPROVED** [review]

File: src/state/lists/hooks.ts [hunk 3]
```diff
@@ -57,10 +58,9 @@ function listToTokenMap(list: TokenList): TokenAddressMap {
   const map = list.tokens.reduce<TokenAddressMapWriteable>((tokenMap, tokenInfo) => {
     const formattedAddress = getFormattedAddress(tokenInfo.address)
 
-    if (tokenMap[tokenInfo.chainId][formattedAddress] !== undefined) {
+    if (tokenMap[tokenInfo.chainId][formattedAddress] !== undefined || !isAddress(tokenInfo.address)) {
```
Comment (viet-nv) [inline src/state/lists/hooks.ts:61]: if tokenInfo.address isnt a valid address, then formattedAddress will be "", right?

Comment (namgold) [reply src/state/lists/hooks.ts:61]: yé

Comment (namgold) [reply src/state/lists/hooks.ts:61]: but merged already :joy:

Comment (viet-nv) [reply src/state/lists/hooks.ts:61]: hmm

**Pull request merged**